### ISSUE TITLE
Other Transport Methods

### ISSUE THREAD

**Issue by Nutomic** [Nutomic/ensichat#25]
- wifi: we can discover other devices using Bonjour or similar and connect to them (when multiple devices are connected to the same hotspot)

- internet: probably with a dht for discovery

- LTE Direct (once that is available for consumers)

**Comment by deavmi** [comment]
It would be cooler if ad-hoc wifi could be implemented. Which would require root access.

**Comment by Nutomic** [comment]
Ad-hoc wifi would basically serve the same purpose as Bluetooth, while using more battery. So I don't really see the advantage there.

**Comment by deavmi** [comment]
Decentralization, as compared to relying on the wifi from a router in a star network.

**Comment by deavmi** [comment]
I say this as it takes the whole decentralized part of the messenger away.

**Comment by deavmi** [comment]
It is a cool idea though, but I would, in my opinion add it as an additional feature to the app rather than a main feature.

**Comment by Nutomic** [comment]
In that case, you can just use Bluetooth for the same effect. Or open a hotspot on any device, and let the other devices connect to it.

**Comment by deavmi** [comment]
@Nutomic I guess so :happy:

**Comment by deavmi** [comment]
Encryption will be on all of these transport methods right? End to end public key encryption? :smile:

**Comment by Nutomic** [comment]
Yes of course ;)

**Comment by deavmi** [comment]
@Nutomic Wow, that's awesome.

![](https://pbs.twimg.com/profile_images/378800000822867536/3f5a00acf72df93528b6bb7cd0a4fd0c.jpeg)

**Comment by deavmi** [comment]
I think the linking of the meshes over the Internet is an awesome idea.

**Comment by deavmi** [comment]
This is turning out to be the cjdns of Bluetooth.

**Comment by deavmi** [comment]
Ad-hoc wifi could still be a cool feature. Just saying.

**Comment by twzkxkan** [comment]
the advantage of wifi is increased range and the use of external antennas

**Comment by deavmi** [comment]
I  assuming the FCC grants wifi spectrum to operate at higher power then.

**Comment by deavmi** [comment]
Bluetooth antennas do not exist for connection via coax ?

**Comment by Nutomic** [comment]
Messaging over internet is now supported in 0.2.0!

I'll close this issue, as wifi doesn't really give us an advantage over bluetooth. And we'll open new issues for other transports (if needed).

**Comment by deavmi** [comment]
YAY! How does one set this up?

**Comment by Nutomic** [comment]
Just start the app, it will connect to the default server automatically. The server will forward messages to other connected devices. Basically like whatsapp :)

**Comment by deavmi** [comment]
Awesome. Can one run a server?

**Comment by Nutomic** [comment]
Yeah, the binary is in the release, and you can change the app settings to connect to it. I'll add proper documentation for that later. Servers also connect to each other to forward messages.

**Comment by deavmi** [comment]
Awesome. :)

**Comment by Nutomic** [comment]
I've added some info on the [wiki](https://github.com/Nutomic/ensichat/wiki). Let me know if anything is unclear or could be improved ;)

**Comment by deavmi** [comment]
Will do. :)

**Comment by deavmi** [comment]
Is your server up? As Scala is getting caight in some execeptions. It can't connect to the machine's IP.

**Comment by Nutomic** [comment]
Looks like it's down right now. I'll check it today or tomorrow.

**Comment by deavmi** [comment]
Have you added exception handling for it? As that would be abgood idea. :)

**Comment by deavmi** [comment]
Okay @nutomic

**Comment by Nutomic** [comment]
Yeah, though I might have missed it somewhere. But there's nothing in the logs, so I'm not sure what's happening. I guess I'll just write a script to restart the server automatically.

**Comment by deavmi** [comment]
Okay. sounds good. :)

**Comment by Nutomic** [comment]
I pushed a new version today that fixes some possible crashes. Let's see how that goes :D

**Comment by deavmi** [comment]
Okay
:)

**Comment by deavmi** [comment]
It's working. My node (server) connected to yours sucessfully.

**Comment by deavmi** [comment]
Is EnsiChat got this upstream network code added in that it can get messages from your node (bbotstrap node)?

**Comment by Nutomic** [comment]
I don't understand your question, can you rephrase it?
